lib/ansible/modules/files/lineinfile.py

```diff
@@ -251,9 +254,12 @@ def present(module, dest, regexp, line, insertafter, insertbefore, create,
                 module.fail_json(msg='Error creating %s Error code: %s Error description: %s' % (b_destpath, e[0], e[1]))
 
         b_lines = []
-    else:
-        with open(b_dest, 'rb') as f:
-            b_lines = f.readlines()
+
+    if not module.check_mode:
+        flock.set_lock(dest)
```

Comment:
It's better to do this as a context manager so it gets released once we leave the context manager's scope.

Since we read the file outside of check mode, we probably want to lock outside of check mode as well.

Reply:
My feeling is that we only need to lock a file if we read with the intention of rewriting that file. So that others don't read an older version for writing after someone else has written. We don't mind if other instances read older versions, because that's just timing.

Reply:
I wrapped present() and absent() in a context manager before, but inside main() which fails, because sometimes the destination path haven't been created yet. I thought about using a context manager from within the functions, it would work but nearly everything inside the function(s) will be wrapped inside of it. Also it would put a lock on the file even when running with --check.
The lock will be released automatically when the process dies, unlock isn't necessary. I agree it's better to do it with a context manager, here however it's inconvenient.